### PR TITLE
Require component-based file names in `assets/js/components`.

### PR DESCRIPTION
**Pull request by benbowler**
## Summary

Require component-based file names in `assets/js/components`.
Addresses issue #2260

## Relevant technical choices

N/A

## Checklist

- [x] My code is tested and passes existing unit tests.
- [x] My code has an appropriate set of unit tests which all pass.
- [x] My code is backward-compatible with WordPress 4.7 and PHP 5.6.
- [x] My code follows the [WordPress](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) coding standards.
- [x] My code has proper inline documentation.
- [x] I have added a QA Brief on the issue linked above.
- [x] I have signed the Contributor License Agreement (see <https://cla.developers.google.com/>).

### CODE REVIEW

**Comment by benbowler** [comment]
I still need to apply the new rule to the files on the linked tickets when unblocked.

**Comment by benbowler** [comment]
Because this is blocked by #2255 I've completed the work and opened the PR against the branch for that ticket. It's a cursory change so it shouldn't prevent it from being reviewed but *we should wait to merge it until #2255 is merged*.

File: .eslintrc.json
Comment (tofumatt) [reply .eslintrc.json:78]: I added `"assets/js/components/**/*.js",` to the list of files to lint and got only 44 errors, all of which are expected (it intelligently lints `index.js` as-expected.)

Is there another issue for dealing with components in folders? If so and we want to wait until that one then okay, but otherwise it makes sense to me for this to include folders (eg `assets/js/components/legacy-setup/site-verification.js` should be `assets/js/components/legacy-setup/SiteVerification.js`, etc.)

**Review by tofumatt — CHANGES_REQUESTED** [review]
I think it would make sense to include other files that violate this rule as well, what do you think?

File: .eslintrc.json
Comment (benbowler) [reply .eslintrc.json:95]: This wasn't intensional, I'll revert.

File: .eslintrc.json
Comment (benbowler) [reply .eslintrc.json:78]: Hey Matt, that covers a lot more cases, I'll update and review the new linter issues that come up.

Comment (benbowler) [reply .eslintrc.json:78]: Also, I have to make sure the imports are updated to the new name as well.

Comment (benbowler) [reply .eslintrc.json:78]: Hey @tofumatt,

Working through one of the issues I can see is:

```
/google-site-kit/assets/js/components/data/index.js
  22:1  error  The directory 'data' must be named 'dataAPI', after the exported value of its index file  filenames/match-exported
```
Do we want to rename the `data` directory to `dataAPI`?

Also, there are a lot of changes in the `legacy-` prefixed folders. Is it worth adding an exclusion to these as I'm assuming they're planning to be phased out over time so it would be unnecessary to rename and bulk update imports for them?

Comment (tofumatt) [reply .eslintrc.json:78]: Ah, I see. Let's not rename `data`, it's arguably legacy too and best to leave alone (it likely has a lot of imports).

Excluding the `legacy-` folders definitely makes sense, exactly as you say. Let's exclude them and the `data` folder, but everything else makes sense to update. 👍🏻

Comment (benbowler) [reply .eslintrc.json]: In the end, I went in and worked through as many files as I could. git is a bit finicky with changing the case of existing file and folder names on a case insensitive filesystem like on the mac so I had a lot of trouble committing the code and husky also cased issues killing the commit. Can you pull down this branch and see if all of the changes happen correctly?

File: assets/js/components/higherorder/withData.js~Stashed changes
Comment (tofumatt) [inline assets/js/components/higherorder/withData.js~Stashed changes:1]: Looks like this file was a stashed change that was committed by mistake. Can you remove it please? 🙂

**Review by tofumatt — CHANGES_REQUESTED** [review]
Looks like this was pushed with some stashed changes and isn't quite complete—originally I thought I'd just delete the `~Stashed changes` files but it seems more than that. Could you take another look?

File: assets/js/components/higherorder/withData.js~Stashed changes
Comment (benbowler) [reply assets/js/components/higherorder/withData.js~Stashed changes:1]: Yes, will do, eslint failed several time trying to commit the file capitalisation changes. I'll reset and try in stages.

Comment (benbowler) [reply assets/js/components/higherorder/withData.js~Stashed changes]: This is failing a lot of the GitHub actions which it's odd because locally these file changes pass the linter. Does the merge successfully update files to the correct capitalisation on your system?

**Comment by benbowler** [comment]
All ready to test @tofumatt

**Review by tofumatt — APPROVED** [review]
Looks good to me. I know this is a big change but I think it makes things more usable and more in-line with what React devs would expect. 👍🏻

**Review by felixarntz — APPROVED** [review]
Awesome!

**Pull request merged**
